kafkazk/partitions.go

```diff
@@ -18,6 +18,26 @@ type Partition struct {
 	Replicas  []int  `json:"replicas"`
 }
 
+// Equal defines equalty between two Partition objects
+// as an equality of topic, partition and replicas.
+func (p Partition) Equal(p2 Partition) bool {
```

Comment:
There's an `equal` method for full partition maps that also includes a repetition of this logic at https://github.com/DataDog/kafka-kit/blob/master/kafkazk/partitions.go#L576. No action needed now since the partition equality check returns both a bool and error, but I want to rewrite the errors, anyway.

We should also write a test for this `Equal` method, though.